sources/loader/main.c

```diff
@@ -101,33 +92,26 @@ void loader_boot(LoaderEntry const *entry)
 {
     VmmSpace vmm = memory_create();
 
-    log$("Loading kernel...");
-
     EntryPointFn entry_point = loader_load_kernel(entry->kernel, vmm);
 
-    log$("Kernel loaded, jumping in to it...");
     Handover *handover = allocate_handover(vmm);
-    efi_populate_handover(entry, handover);
 
-    efi_deinit();
+    loader_populate_handover(entry, handover);
+
     memory_switch(vmm);
```

Comment:
The memory switch should be the last thing we do before jumping into the kernel.